tests/ClientBuilderTest.php

```diff
@@ -96,14 +103,17 @@ public function testSetMessageFactory(): void
      */
     public function testSetTransport(): void
     {
-        /** @var TransportInterface|MockObject $transport */
+        /** @var TransportInterface&MockObject $transport */
         $transport = $this->createMock(TransportInterface::class);
+        $transport->expects($this->once())
+            ->method('send')
+            ->willReturn('ddb4a0b9ab1941bf92bd2520063663e3');
 
-        $clientBuilder = ClientBuilder::create(['dsn' => 'http://public:secret@example.com/sentry/1']);
-        $clientBuilder->setTransport($transport);
+        $client = ClientBuilder::create(['dsn' => 'http://public@example.com/sentry/1'])
+            ->setTransport($transport)
+            ->getClient();
 
-        $this->assertAttributeSame($transport, 'transport', $clientBuilder);
-        $this->assertAttributeSame($transport, 'transport', $clientBuilder->getClient());
+        $client->captureMessage('foo');
```

Comment:
Whenever possible, I would add an assert to check the mocked value, like here with the event ID.